### DISCUSSION TITLE
Why doesn't Headless Testing wait for the task to return the result when using await?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15100] [Q&A] [answered]
Followed this : https://github.com/AvaloniaUI/Avalonia/discussions/15050#discussioncomment-8849697

`using var session = HeadlessUnitTestSession.StartNew(typeof(App));

await session.Dispatch(async Task () =>
{
	  var result = await test();    //wrong
}, CancellationToken.None);`

Everything is ok in tests that do not require waiting. But cannot get the result correctly if using await, because it does not wait for the task. Did I do something wrong?

**Comment by maxkatz6** [comment, marked as answer]
I guess it uses `Dispatch(Func<TResult>)` overload.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Headless/Avalonia.Headless/HeadlessUnitTestSession.cs#L44-L68

Can you return some value from the session? Like, `return true;` in the end? So it will force compiler to pick `Func<Task<TResult>>` overload instead.
